src/rgw/rgw_bucket.cc

```diff
@@ -2071,7 +2071,9 @@ void RGWDataChangesLog::read_clear_modified(map<int, set<string> > &modified)
 
 void RGWBucketCompleteInfo::dump(Formatter *f) const {
   encode_json("bucket_info", info, f);
-  encode_json("attrs", attrs, f);
+  map<string, bufferlist> encode_attrs = attrs;
+  encode_attrs.erase(RGW_ATTR_LC);
+  encode_json("attrs", encode_attrs, f);
```

Comment:
@cbodley , it can be confusing. How can an admin know it has no effect in this zone?

Reply:
given the other change to erase the RGW_ATTR_LC attribute in `RGWBucketCompleteInfo::decode_json()`, we won't store that lifecycle config on other zones. so if a lifecycle config is present in `radosgw-admin metadata get` on a zone, that means it is in effect there

Reply:
@cbodley 
I have took a look at codes of 'radosgw-admin metadata get', it doesn't interact with other zones and just obtain local bucket info, so it may have nothing to do with 'RGWBucketCompleteInfo::decode_json()'

Reply:
sorry for confusing things. your current approach filters this `RGW_ATTR_LC` attribute out of the bucket instance metadata before it gets sent to other zones. this has the side effect that `radosgw-admin metadata get` on the zone that set this lifecycle config won't see it either. i'm just suggesting that we do the filtering on decode instead, so that other zones will ignore non-local lifecycle configs

Reply:
Thanks for your detail explanation.
I think dump RGW_ATTR_LC by `radosgw-admin metadata get` or inject RGW_ATTR_LC by `radosgw-admin metadata put` may be a bit ideal, because it's not human readable at all.
so I'm thinking why we need RGW_ATTR_LC from `radosgw-admin metadata get`, the only reason I can get is that we can use it to do `radosgw-admin metadata put`.
Let's focus on `radosgw-admin metadata put`, it will call `RGWBucketInstanceMetadataHandler::put` finally, which is same as meta sync. So things will be more complex if we inject lc configuration by `radosgw-admin metadata put`, which codes can't tell from multisite meta sync.

Reply:
Of course, I agree you that it's not a good way to do exceptional things in RGWBucketCompleteInfo::dump